orion/primitives/units.py

```diff
@@ -0,0 +1,910 @@
+
+    Args:
+        window_size (int):
+            Window size of each sample. Default to 250.
+        pred_len (int):
+            Prediction horizon length. Default to 1.
+        prompt_num (int):
+            Number of prompt tokens. Default to 10.
+        d_model (int):
+            Model dimension. Default to 64.
+        patch_len (int):
+            Length of each patch. Default to 1.
+        step (int):
+            Stride length between samples. Default to 1.
+        dropout (int):
+            Dropout value of the network. Default to 0.1.
+        e_layers (int):
+            Number of embedding layers. Default to 3.
+        n_heads (int):
+            Number of heads. Default to 8.
+    """
+
+    def __init__(self,
+                 window_size=250,
+                 pred_len=1,
+                 prompt_num=10,
+                 d_model=64,
+                 patch_len=1,
+                 step=1,
+                 dropout=0.1,
+                 e_layers=3,
+                 n_heads=8):
+        super(UniTS, self).__init__()
+
+        self.window_size = window_size
+        self.pred_len = pred_len
+        self.prompt_num = prompt_num
+        self.d_model = d_model
+        self.patch_len = patch_len
+        self.step = step
+        self.dropout = dropout
+        self.e_layers = e_layers
+        self.n_heads = n_heads
+
+        self.model = self._build_model()
+
+    def _build_model(self):
+        model = Model(window_size=self.window_size,
+                      pred_len=self.pred_len,
+                      prompt_num=self.prompt_num,
+                      d_model=self.d_model,
+                      patch_len=self.patch_len,
+                      step=self.step,
+                      dropout=self.dropout,
+                      e_layers=self.e_layers,
+                      n_heads=self.n_heads)
+        return model
+
+    def predict(self, X, index):
+        """Forecasting timeseries
+        Args:
+            X (ndarray):
+                input timeseries.
+            index (ndarray
+                timestamps array.
+        Return:
+            ndarray, ndarray, ndarray:
+                * forecasted timeseries.
+                * array of truncated ground truth with same size as forecasted timeseries.
+                * array of timestamps with same size as forecasted timeseries.
+        """
+
+        self.test_dataset = Signal(X, index,
+                                   window_size=self.window_size,
+                                   pred_length=self.pred_len)
+
+        first_index = self.test_dataset.first_index()
+
+        self.test_loader = DataLoader(self.test_dataset,
+                                      batch_size=1,
+                                      shuffle=False,
+                                      num_workers=0,
+                                      drop_last=False)
+
+        load_path = UNITS_PATH
+        with smart_open(load_path, 'rb') as f:
```

Comment:
make `load_path` an attribute to `UniTS` class that defaults to `UNITS_PATH`, then call:
```python3
with smart_open(self.load_path, 'rb') as f:
```